source/adapters/native_cpu/threadpool.hpp

```diff
@@ -121,57 +116,31 @@ class worker_thread {
 // parameters and futures.
 class simple_thread_pool {
 public:
-  simple_thread_pool(size_t numThreads = 0) noexcept : m_isRunning(false) {
-    this->resize(numThreads);
-    this->start();
-  }
-
-  ~simple_thread_pool() { this->stop(); }
-
-  // Creates and launches the worker threads
-  inline void start() {
-    if (this->is_running()) {
-      return;
-    }
-    size_t threadId = 0;
-    for (auto &t : m_workers) {
-      t.start(threadId);
-      threadId++;
+  simple_thread_pool() noexcept : m_isRunning(false), m_numThreads(get_num_threads()) {
+    for(size_t i = 0; i < m_numThreads; i++) {
+      m_workers.emplace_front(i);
     }
-    m_isRunning.store(true, std::memory_order_release);
+    m_isRunning.store(true);
   }
 
-  // Waits for all tasks to finish and destroys the worker threads
-  inline void stop() {
+  ~simple_thread_pool() { 
     for (auto &t : m_workers) {
       t.stop();
     }
-    m_isRunning.store(false, std::memory_order_release);
+    m_isRunning.store(false);
   }
 
-  inline void resize(size_t numThreads) {
-    char *envVar = std::getenv("SYCL_NATIVE_CPU_HOST_THREADS");
-    if (envVar) {
-      numThreads = std::stoul(envVar);
-    }
-    if (numThreads == 0) {
-      numThreads = std::thread::hardware_concurrency();
-    }
-    if (!this->is_running() && (numThreads != this->num_threads())) {
-      m_workers = decltype(m_workers)(numThreads);
-    }
-  }
 
   inline void schedule(const worker_task_t &task) {
     // Schedule the task on the best available worker thread
     this->best_worker().schedule(task);
   }
 
   inline bool is_running() const noexcept {
-    return m_isRunning.load(std::memory_order_acquire);
+    return m_isRunning.load();
```

Comment:
The default ordering for atomics is seq-cst, meaning that on x86 there might be an additional fence. The previous implementation with acquire/release ordering has less overhead. Did coverity complain about this?

Reply:
Coverity didn't complain about this, but while working on the changes for this PR I thought that using `seq_cst` may be safer here, I'm aware that this may have more overhead. I'm not very familiar with C++'s memory ordering so I'm open to suggestions

Reply:
In this case I believe acquire/release is safe. When you are stopping, release in store is sufficient to ensure that all writes from the calling thread are visible in the workers. Likewise, when the value is read with acquire, that ensures that all prior writes to this variable from other threads are visible in the worker.

Reply:
I've switched back to acquire/release, thanks for the feedback